ui/js/graphs.js

```diff
@@ -684,20 +684,21 @@ perf.controller('GraphsCtrl', [
             });
 
             modalInstance.result.then(function(series) {
-              series.highlightedPoints = [];
-              series.visible = true;
-              series.color = availableColors.pop();
-
-              $scope.seriesList.push(series);
-              if( !$scope.highlightedRevision ) {
-                $scope.highlightedRevision = '';
-              }
-              if (!$scope.zoom) {
-                $scope.zoom = {};
+              for (var i = 0; i < series.length; i++) {
+                  series[i].hightlightedPoints = [];
+                  series[i].visible = true;
+                  series[i].color = availableColors.pop();
+                  $scope.seriesList.push(series[i]);
+                  if (!$scope.highlightedRevision) {
+                      $scope.highlightedRevision = '';                  
+                  }
+                  if (!$scope.zoom) {
+                      $scope.zoom = {};
+                  }
+                  updateDocument();
```

Comment:
Lines 692-698 should not be in the loop. You should call them after the loop (or the foreach, actually, once you change that) is finished.